### PR TITLE
Make all scale down arguments optional

### PR DESCRIPTION
**Pull request by bison**
This makes all the scale down related arguments properly optional. It also fixes the MachineAutoscaler example, and updates the operator's ClusterRole to allow managing ClusterOperator objects.

### CODE REVIEW

**Comment by bison** [comment]
/test gofmt

This failure was a timeout pulling the base image or something.

File: pkg/controller/clusterautoscaler/clusterautoscaler.go [hunk 1]
```diff
@@ -87,18 +87,31 @@ func AutoscalerArgs(ca *v1alpha1.ClusterAutoscaler, cfg *Config) []string {
...
-		args = append(args, ScaleDownUnneededTimeArg.Value(sd.UnneededTime))
+	if sd.DelayAfterAdd != nil {
+		args = append(args, ScaleDownDelayAfterAddArg.Value(*sd.DelayAfterAdd))
+	}
+
+	if sd.DelayAfterDelete != nil {
+		args = append(args, ScaleDownDelayAfterDeleteArg.Value(*sd.DelayAfterDelete))
+	}
+
+	if sd.DelayAfterDelete != nil {
```
Comment (ingvagabund) [inline pkg/controller/clusterautoscaler/clusterautoscaler.go]: This is duplicated

Comment (bison) [reply pkg/controller/clusterautoscaler/clusterautoscaler.go]: Oops. Thank. Fixed.

**Comment by ingvagabund** [comment]
The third commit is not related to this PR. Thought, it will get merged into the same tree and it's Friday.

/approve

**Pull request merged**